src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1

```diff
@@ -575,4 +573,57 @@ class ServiceConnection: ADOSVTBase
         return $controlResult
     }
     
+    hidden CheckActiveConnection()
+    {
+        try
+        {
+            if ($this.ServiceConnEndPointDetail -and [Helpers]::CheckMember($this.ServiceConnEndPointDetail, "serviceEndpointExecutionHistory") ) 
+            {
+                #if this job is still running then finishTime is not available. pass the control
+                if ([Helpers]::CheckMember($this.ServiceConnEndPointDetail.serviceEndpointExecutionHistory[0].data, "finishTime")) 
+                {
+                    #Get the last known usage (job) timestamp of the service connection
+                    $svcLastRunDate = $this.ServiceConnEndPointDetail.serviceEndpointExecutionHistory[0].data.finishTime;
+                    
+                    #format date
+                    $formatLastRunTimeSpan = New-TimeSpan -Start (Get-Date $svcLastRunDate)
+                    
+                    # $inactiveLimit denotes the upper limit on number of days of inactivity before the svc conn is deemed inactive.
+                    if ($this.ControlSettings -and [Helpers]::CheckMember($this.ControlSettings, "ServiceConnection.ServiceConnectionHistoryPeriodInDays") ) 
+                    {
+                        $inactiveLimit = $this.ControlSettings.ServiceConnection.ServiceConnectionHistoryPeriodInDays
+                        if ($formatLastRunTimeSpan.Days -gt $inactiveLimit)
+                        {
+                            $this.SvcConnActivityDetail.isSvcConnActive = $false;
+                            $this.SvcConnActivityDetail.message = "Service connection has not been used in the last $inactiveLimit days.";
+                        }
+                        else
+                        {
+                            $this.SvcConnActivityDetail.isSvcConnActive = $true;
+                            $this.SvcConnActivityDetail.message =  "Service connection has been used in the last $inactiveLimit days.";
+                        }
+                    }
+                    else {
+                        $this.SvcConnActivityDetail.isSvcConnActive = $false;
+                        $this.SvcConnActivityDetail.message = "History period in days (ServiceConnectionHistoryPeriodInDays) to check last running day of service connection is not defined in your organization policy. Please update your ControlSettings.json as per the latest AzSK.ADO PowerShell module.";
```

Comment:
remove

Reply:
Could not fetch the inactive limit for service connection.

Reply:
message changed